cmd/kubeless/function_test.go

```diff
@@ -217,4 +220,92 @@ func TestGetFunctionDescription(t *testing.T) {
 		t.Error("Unexpected result")
 	}
 
+	// It should upload the function if Minio is available
+	minioFakeSvc := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "kubeless",
+			Name:      "minio",
+		},
+	}
+	minioConfig := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "minio-config",
+			Namespace: "kubeless",
+		},
+		Data: map[string]string{
+			"maxFileSize": "50Mi",
+		},
+	}
+	uploadFakeJob := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "kubeless",
+			Name:      "upload-file",
+		},
+		Status: batchv1.JobStatus{
+			Succeeded: 1,
+		},
+	}
+	cli := &fake.Clientset{}
```

Comment:
Replace this and following `AddReactor` lines with:
```golang
cli := fake.NewSimpleClientset(&minioFakeSvc, &minioConfig, &uploadFakeJob)
```

Reply:
It doesn't work for me that way. When you do `fake.NewSimpleClientset(&uploadFakeJob)` it returns the object for every verb (get, create, update...). That way the action that creates the job fails because the job already exists. With `AddReactor` I control the verb so I can tell it to return the job when doing a `get` but not when doing a `create`.

Reply:
oh interesting, I must have only used that for simple cases and never noticed :P

Reply:
(removed that code so it no longer matters)